src/parser.ts

```diff
@@ -102,6 +102,10 @@ export class Parser extends DiagnosticEmitter {
   donelog: Set<string> = new Set();
   /** Optional handler to intercept comments while tokenizing. */
   onComment: CommentHandler | null = null;
+  /** Current file being parsed. */
+  currentFile: string;
```

Comment:
Would prefer `currentSource: Source` here, so all of the source's information is available in subsequent steps.